pkg/services/rendering/rendering.go

```diff
@@ -110,6 +113,23 @@ func (rs *RenderingService) RenderErrorImage(err error) (*RenderResult, error) {
 }
 
 func (rs *RenderingService) Render(ctx context.Context, opts Opts) (*RenderResult, error) {
+	startTime := time.Now()
+	result, err := rs.render(ctx, opts)
+	elapsedTime := time.Since(startTime).Milliseconds()
+	if status.Code(err) == codes.DeadlineExceeded {
```

Comment:
You want to check for `ErrTimeout` here instead, see https://github.com/grafana/grafana/blob/d0e23af2283a5afdb6b026bb9e30bcf665a8b5ce/pkg/services/rendering/interface.go#L11 and https://github.com/grafana/grafana/blob/d0e23af2283a5afdb6b026bb9e30bcf665a8b5ce/pkg/services/rendering/http_mode.go#L84

The `if status.Code(err) == codes.DeadlineExceeded {` you want to move to `plugin_mode.go` and return `ErrTimeout` instead.